orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/tasks/RunKayentaCanaryTask.kt

```diff
@@ -20,33 +20,66 @@ import com.netflix.spinnaker.orca.ExecutionStatus.SUCCEEDED
 import com.netflix.spinnaker.orca.Task
 import com.netflix.spinnaker.orca.TaskResult
 import com.netflix.spinnaker.orca.ext.mapTo
+import com.netflix.spinnaker.orca.jackson.OrcaObjectMapper
 import com.netflix.spinnaker.orca.kayenta.CanaryExecutionRequest
+import com.netflix.spinnaker.orca.kayenta.CanaryScopes
 import com.netflix.spinnaker.orca.kayenta.KayentaService
 import com.netflix.spinnaker.orca.kayenta.model.RunCanaryContext
 import com.netflix.spinnaker.orca.pipeline.model.Stage
-import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Component
 import java.util.Collections.singletonMap
 
 @Component
 class RunKayentaCanaryTask(
   private val kayentaService: KayentaService
 ) : Task {
-
-  private val log = LoggerFactory.getLogger(javaClass)
-
   override fun execute(stage: Stage): TaskResult {
     val context = stage.mapTo<RunCanaryContext>()
+
+    val scopes = stage.deployedCanaryClusters?.let { deployedClusters ->
+      // Assume for now that if we deployed canary clusters that
```

Comment:
This makes sense if the plan is to hide the `locations/scopes` controls in the stage config when the stage is also handling the provisioning.